Src/AutoFixture.xUnit2/MemberAutoDataAttribute.cs

```diff
@@ -67,5 +74,31 @@ protected MemberAutoDataAttribute(DataAttribute autoDataAttribute, string member
         /// Gets the parameters passed to the member. Only supported for static methods.
         /// </summary>
         public IEnumerable<object> Parameters { get; }
+
+        /// <summary>
+        /// Returns the composition of test data from the member on the test class
+        /// combined with the auto-generated data specimens generated by AutoFixture.
+        /// </summary>
+        /// <param name="testMethod">The method that is being tested.</param>
+        /// <returns>
+        /// Returns the composition of the theory data provided by MemberDataAttribute and AutoFixture.
+        /// </returns>
+        public override IEnumerable<object[]> GetData(MethodInfo testMethod)
+        {
+            if (testMethod == null) throw new ArgumentNullException(nameof(testMethod));
+
+            var memberData = this.MemberDataAttribute.GetData(testMethod).ToList();
+            var autoData = this.AutoDataAttribute.GetData(testMethod).FirstOrDefault();
+
+            if (autoData == null) throw new InvalidOperationException($"{nameof(autoData)} cannot be null");
+
+            var combinedDataSet = new List<IEnumerable<object[]>>
+            {
+                memberData,
+                Enumerable.Repeat(autoData, memberData.Count)
+            };
+
+            return combinedDataSet.Zip(dataSet => dataSet.Collapse().ToArray());
```

Comment:
This looks good, but I feel it can be made a bit more readable.

If you add an overload of the `Collapse` method that looks like this:

```csharp
internal static IEnumerable<T> Collapse<T>(this IEnumerable<T> sequence, params IEnumerable<T>[] others)
{
    return Collapse(new[] { sequence }.Concat(others));
}
```

Then, you can combine the values from the `MemberDataAttribute` with the ones generated by AutoFixture with a single LINQ expression:

```suggestion
            var memberData = this.MemberDataAttribute.GetData(testMethod);
            var autoData = this.AutoDataAttribute.GetData(testMethod).Take(1);

            if (autoData == null) throw new InvalidOperationException($"{nameof(autoData)} cannot be null");

            return from memberValues in memberData
                   from autoValues in autoData
                   let combinedValues = memberValues.Collapse(autoValues)
                   select combinedValues.ToArray();
```

This is simply the Cartesian product between the rows provided by `MemberDataAttribute` and the first row provided by `AutoDataAttribute`.

Reply:
@ecampidoglio there's a minor oversight in the proposed implementation, `autoData` can never be `null`. In case the source collection is empty `Skip()` and `Take()` yield empty enumerables as well.
I'd prefer the second option, it doesn't use custom expressions.

Reply:
@aivascu True that, we can remove the null check altogether. 👍